src/main/java/io/weaviate/client/v1/rbac/model/RbacAction.java

```diff
@@ -0,0 +1,45 @@
+package io.weaviate.client.v1.rbac.model;
+
+/**
+ * RbacAction is a utility interface to allow retrieving concrete enum values
+ * from their underlying string representations.
+ *
+ * <p>
+ * Usage:
+ *
+ * <pre>
+ * enum MyAction implements RbacAction {
+ *   FOO("do_foo"),
+ *   BAR("do_bar");
+ *
+ *   @Getter
+ *   private String value;
+ * }
+ * </pre>
+ *
+ * <p>
+ * Then {@code MyAction.FOO} can be retrieved from "do_foo" using
+ * {@link #fromString}.
+ */
+interface RbacAction {
```

Comment:
shouldn't RbacAction interface be public?

Reply:
It doesn't need to, because we only ever use it internally for deserializing responses from the server.
What the user passes/receives is one of the "concrete" public enums that implement `RbacAction`.